File: c/3.c
int lengthOfLongestSubstring(char* s) {
    int n = strlen(s), p = 0, l = 0;
    int m[128];
    for (int i = 0; i < 128; i++)
        m[i] = -1;
    for (int i = 0; i < n; i++) {
        p = fmax(p, m[s[i]] + 1);
        m[s[i]] = i;
        l = fmax(l, i - p + 1);
    }
    return l;
}

File: javascript/3.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function (s) {
    let n = s.length, p = 0, l = 0, m = new Array(128).fill(-1);
    for (let i = 0; i < n; i++) {
        p = Math.max(p, m[s.codePointAt(i)] + 1);
        m[s.codePointAt(i)] = i;
        l = Math.max(l, i - p + 1);
    }
    return l;
};
